src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1314 @@
+// This file is part of the uutils coreutils package.
+//
+// (c) Tyler Steele <tyler.steele@protonmail.com>
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+// spell-checker:ignore (T0DO)
+
+#[macro_use]
+extern crate uucore;
+use uucore::InvalidEncodingHandling;
+
+#[cfg(test)]
+mod dd_unit_tests;
+
+mod datastructures;
+use datastructures::*;
+
+mod parseargs;
+use parseargs::Matches;
+
+mod conversion_tables;
+use conversion_tables::*;
+
+use byte_unit::Byte;
+use clap::{self, crate_version};
+use debug_print::debug_println;
+use gcd::Gcd;
+#[cfg(unix)]
+use libc;
+use signal_hook::consts::signal;
+use std::cmp;
+use std::convert::TryInto;
+use std::env;
+use std::error::Error;
+use std::fs::{File, OpenOptions};
+use std::io::{self, Read, Seek, Write};
+#[cfg(unix)]
+use std::os::unix::fs::OpenOptionsExt;
+use std::sync::{atomic::AtomicUsize, atomic::Ordering, mpsc, Arc};
+use std::thread;
+use std::time;
+
+const SYNTAX: &str = "dd [OPERAND]...\ndd OPTION";
+const ABOUT: &str = "copy, and optionally convert, a file system resource";
+const BUF_INIT_BYTE: u8 = 0xDD;
+const RTN_SUCCESS: i32 = 0;
+const RTN_FAILURE: i32 = 1;
+
+struct Input<R: Read> {
+    src: R,
+    non_ascii: bool,
+    ibs: usize,
+    xfer_stats: Option<StatusLevel>,
+    count: Option<CountType>,
+    cflags: IConvFlags,
+    iflags: IFlags,
+}
+
+impl Input<io::Stdin> {
+    fn new(matches: &Matches) -> Result<Self, Box<dyn Error>> {
+        let ibs = parseargs::parse_ibs(matches)?;
+        let non_ascii = parseargs::parse_input_non_ascii(matches)?;
+        let xfer_stats = parseargs::parse_status_level(matches)?;
+        let cflags = parseargs::parse_conv_flag_input(matches)?;
+        let iflags = parseargs::parse_iflags(matches)?;
+        let skip = parseargs::parse_skip_amt(&ibs, &iflags, matches)?;
+        let count = parseargs::parse_count(&iflags, matches)?;
+
+        let mut i = Input {
+            src: io::stdin(),
+            non_ascii,
+            ibs,
+            xfer_stats,
+            count,
+            cflags,
+            iflags,
+        };
+
+        if let Some(amt) = skip {
+            let mut buf = vec![BUF_INIT_BYTE; amt];
+
+            i.force_fill(&mut buf, amt)?;
+        }
+
+        Ok(i)
+    }
+}
+
+fn make_unix_iflags(oflags: &IFlags) -> Option<libc::c_int> {
+    let mut flag = 0;
+
+    if oflags.direct {
+        flag |= libc::O_DIRECT;
```

Comment:
please fix this error

Reply:
the CI was showing error (didn't exist)
you will be able to see them in `Files changed` once the CI is over